src/main/java/seedu/eylah/diettracker/model/self/Bmi.java

```diff
@@ -0,0 +1,88 @@
+package seedu.eylah.diettracker.model.self;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.eylah.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents the BMI in the diet tracker of EYLAH.
+ * Guarantees: immutable; is valid as declared in {@link #isValidHeight(String)}
+ */
+public class Bmi {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Please input a height and a weight to calculate BMI";
+
+    /*
+     * The first character of the height must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String VALIDATION_REGEX = "[\\p{Alnum}]*";
+
+    public final Float height;
+    public final Float weight;
+    public final String bmi;
+    public final Float bmiFloat;
+
+    /**
+     * Constructs a {@code Height}.
+     *
+     * @param height A valid height.
+     */
+    public Bmi(Height height, Weight weight) {
+        requireNonNull(height);
+        checkArgument(isValidHeight(height.toString()), MESSAGE_CONSTRAINTS);
+        checkArgument(isValidWeight(weight.toString()), MESSAGE_CONSTRAINTS);
+
+        this.height = height.getHt();
+        this.weight = weight.getWt();
+        this.bmiFloat = calcBmi();
+        this.bmi = this.bmiFloat.toString();
+    }
+
+    /**
+     * Returns true if a given string is a valid height.
+     */
+    public static boolean isValidHeight(String test) {
+        return test.matches(VALIDATION_REGEX);
+    }
+
+    /**
+     * Returns true if a given string is a valid height.
+     */
+    public static boolean isValidWeight(String test) {
+        return test.matches(VALIDATION_REGEX);
+    }
+
+    /**
+     * Method to calculate BMI. Formula used to calculate BMI is
+     * Weight (in kilograms) divided by Height^2 (in metres^2).
+     * @return Float of the BMI value.
+     */
+    private Float calcBmi() {
+        // Weight/Height^2
```

Comment:
Perhaps this comment can be moved up and explained in the javaDoc above it?

Reply:
Yes agree, but maybe keep the comment but copy and paste the formula up in the javadoc for the class to just give an overview.

Reply:
It is explained in the Javadoc for the method already.